File: backend-nodejs/models/Product.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Product = sequelize.define('Product', {
  id: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    autoIncrement: true
  },
  name: {
    type: DataTypes.STRING(200),
    allowNull: false
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  basePrice: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    field: 'base_price'
  },
  discountPercent: {
    type: DataTypes.DECIMAL(5, 2),
    defaultValue: 0,
    field: 'discount_percent'
  },
  discountedPrice: {
    type: DataTypes.DECIMAL(10, 2),
    field: 'discounted_price'
  },
  unitsInStock: {
    type: DataTypes.INTEGER,
    allowNull: false,
    field: 'units_in_stock'
  },
  imageUrl: {
    type: DataTypes.STRING(500),
    allowNull: true,
    field: 'image_url'
  },
  brand: {
    type: DataTypes.STRING(100),
    allowNull: true
  },
  sizes: {
    type: DataTypes.STRING(100),
    allowNull: true
  },
  averageRating: {
    type: DataTypes.DOUBLE,
    defaultValue: 0.0,
    field: 'average_rating'
  },
  totalReviews: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    field: 'total_reviews'
  },
  categoryId: {
    type: DataTypes.BIGINT,
    allowNull: false,
    field: 'category_id'
  },
  createdDate: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
    field: 'created_date'
  },
  popularityScore: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    field: 'popularity_score'
  }
}, {
  tableName: 'products',
  timestamps: false
});

module.exports = Product;

File: backend-dotnet/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ClothingStore.Models
{
    [Table("products")]
    public class Product
    {
        [Column("id")]
        public long Id { get; set; }
        
        [Column("name")]
        [Required]
        public string Name { get; set; } = string.Empty;
        
        [Column("description")]
        public string? Description { get; set; }
        
        [Column("base_price")]
        [Required]
        public decimal BasePrice { get; set; }
        
        [Column("discounted_price")]
        public decimal? DiscountedPrice { get; set; }
        
        [Column("discount_percent")]
        public decimal DiscountPercent { get; set; }
        
        [Column("units_in_stock")]
        [Required]
        public int UnitsInStock { get; set; }
        
        [Column("brand")]
        public string? Brand { get; set; }
        
        [Column("sizes")]
        public string? Sizes { get; set; }
        
        [Column("image_url")]
        public string? ImageUrl { get; set; }
        
        [Column("category_id")]
        [Required]
        public long CategoryId { get; set; }
    }
}